pkg/controller/compute/v1beta1/cluster.go

```diff
@@ -0,0 +1,287 @@
+				}
+				break
+			}
+		}
+	}
+
+	switch cr.Status.AtProvider.Status {
+	case v1beta1.ClusterStateRunning:
+		cr.Status.SetConditions(v1alpha1.Available())
+		resource.SetBindable(cr)
+	case v1beta1.ClusterStateProvisioning:
+		cr.Status.SetConditions(v1alpha1.Creating())
+	case v1beta1.ClusterStateUnspecified, v1beta1.ClusterStateDegraded, v1beta1.ClusterStateError, v1beta1.ClusterStateReconciling:
+		cr.Status.SetConditions(v1alpha1.Unavailable())
+	}
+
+	upToDate, _ := gke.IsUpToDate(&cr.Spec.ForProvider, *existing)
+
+	return resource.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  upToDate,
+		ConnectionDetails: connectionDetails(existing),
+	}, nil
+}
+
+func (e *clusterExternal) Create(ctx context.Context, mg resource.Managed) (resource.ExternalCreation, error) {
+	i, ok := mg.(*v1beta1.GKECluster)
+	if !ok {
+		return resource.ExternalCreation{}, errors.New(errNotCluster)
+	}
+
+	// Generate GKE cluster from resource spec.
+	cluster := gke.GenerateCluster(i.Spec.ForProvider)
+
+	// Insert default node pool for bootstrapping cluster.
+	gke.GenerateNodePoolForCreate(cluster)
+
+	create := &container.CreateClusterRequest{
+		Cluster: cluster,
+	}
+
+	parent := fmt.Sprintf(parentFormat, e.projectID, i.Spec.ForProvider.Location)
+
+	if _, err := e.cluster.Projects.Locations.Clusters.Create(parent, create).Context(ctx).Do(); err != nil {
+		return resource.ExternalCreation{}, errors.Wrap(err, errCreateCluster)
+	}
+
+	clusterName := fmt.Sprintf(clusterNameFormat, e.projectID, i.Spec.ForProvider.Location, i.Spec.ForProvider.Name)
+	meta.SetExternalName(i, clusterName)
+
+	// TODO(hasheddan): go ahead and propagate username / password here if set in spec?
+	return resource.ExternalCreation{}, nil
+}
+
+func (e *clusterExternal) Update(ctx context.Context, mg resource.Managed) (resource.ExternalUpdate, error) {
+	i, ok := mg.(*v1beta1.GKECluster)
+	if !ok {
+		return resource.ExternalUpdate{}, errors.New(errNotCluster)
+	}
+
+	// We have to get the cluster again here
+	existing, err := e.cluster.Projects.Locations.Clusters.Get(meta.GetExternalName(mg)).Context(ctx).Do()
+	if err != nil {
+		return resource.ExternalUpdate{}, errors.Wrap(err, errGetCluster)
+	}
+
+	upToDate, updateFn := gke.IsUpToDate(&i.Spec.ForProvider, *existing)
+	if upToDate {
+		return resource.ExternalUpdate{}, nil
+	}
+
+	_, err = updateFn(e.cluster, ctx, meta.GetExternalName(mg))
+	return resource.ExternalUpdate{}, errors.Wrap(err, errUpdateCluster)
+}
+
+func (e *clusterExternal) Delete(ctx context.Context, mg resource.Managed) error {
+	i, ok := mg.(*v1beta1.GKECluster)
+	if !ok {
+		return errors.New(errNotCluster)
+	}
+	i.SetConditions(runtimev1alpha1.Deleting())
+
+	_, err := e.cluster.Projects.Locations.Clusters.Delete(meta.GetExternalName(mg)).Context(ctx).Do()
+	if gcp.IsErrorNotFound(err) {
+		return nil
+	}
+	return errors.Wrap(err, errDeleteCluster)
```

Comment:
Nitpick: This could be a one liner like the below.

```go
_, err := // e.cluster...
return errors.Wrap(resource.Ignore(gcp.IsErrorNotFound, err), errDeleteCluster
```